### DISCUSSION TITLE
useRevalidator and shouldRevalidate

### DISCUSSION THREAD

**Discussion by nzlotrfan** [remix-run/react-router#11417] [Q&A]
Hi, I'm using createBrowserRouter and I notice that if I have shouldRevalidate set to false (`shouldRevalidate: () => false,`) for a route's loader, I am unable to manually trigger a revalidation of a loader's data with `revalidator.revalidate();` using the useRevalidator hook. Is this the intended behaviour? I don't want my route to automatically revalidate because I'm using searchParams and it leads to multiple firings of the loader, but then if I disable it the revalidate fn just doesn't execute.

**Comment by bakikucukcakiroglu** [comment]
Yes, I'm having the same issue. Any workaround?

**Reply by nzlotrfan** [reply]
I ended up using Tanstack Query instead.

**Reply by bakikucukcakiroglu** [reply]
> I ended up using Tanstack Query instead.

Wow!
